pkg/querier/lazyquery/lazyquery.go

```diff
@@ -48,22 +48,17 @@ func (l LazyQuerier) createSeriesSet(selectSorted bool, params *storage.SelectHi
 	// waiting for the result yet (or anymore).
 	future := make(chan storage.SeriesSet, 1)
 	go func() {
-		set, _, err := l.next.Select(selectSorted, params, matchers...)
-		if err != nil {
-			future <- errSeriesSet{err}
-		} else {
-			future <- set
-		}
+		future <- l.next.Select(selectSorted, params, matchers...)
 	}()
 	return future
 }
 
 // Select implements Storage.Querier
-func (l LazyQuerier) Select(selectSorted bool, params *storage.SelectHints, matchers ...*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
+func (l LazyQuerier) Select(selectSorted bool, params *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
```

Comment:
We can get rid of `createSeriesSet` now. It was introduced when there were Select and SelectSorted methods.

Reply:
Good point, done.